bot/exts/utilities/twemoji.py

```diff
@@ -0,0 +1,148 @@
+import logging
+import re
+from typing import Literal
+
+import discord
+from discord.ext import commands
+from emoji import UNICODE_EMOJI_ENGLISH, is_emoji
+
+from bot.bot import Bot
+from bot.constants import Colours
+from bot.utils.extensions import invoke_help_command
+
+log = logging.getLogger(__name__)
+BASE_URLS = {
+    "png": "https://raw.githubusercontent.com/twitter/twemoji/master/assets/72x72/",
+    "svg": "https://raw.githubusercontent.com/twitter/twemoji/master/assets/svg/",
+}
+CODE = re.compile(r"[a-f1-9][a-f0-9]{3,5}$")
+
+
+class Twemoji(commands.Cog):
+    """Utilities for working with Twemojis."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @staticmethod
+    def get_url(codepoint: str, format: Literal["png", "svg"]) -> str:
+        """Returns a source file URL for the specified Twemoji, in the corresponding format."""
+        return f"{BASE_URLS[format]}{codepoint}.{format}"
+
+    @staticmethod
+    def alias_to_name(alias: str) -> str:
+        """
+        Transform a unicode alias to an emoji name.
+
+        Example usages:
+        >>> alias_to_name(":falling_leaf:")
+        "Falling leaf"
+        >>> alias_to_name(":family_man_girl_boy:")
+        "Family man girl boy"
+        """
+        name = alias[1:-1].replace("_", " ")
+        return name.capitalize()
+
+    @staticmethod
+    def build_embed(codepoint: str) -> discord.Embed:
+        """Returns the main embed for the `twemoji` commmand."""
+        emoji = "".join(Twemoji.emoji(e) for e in codepoint.split("-"))
+
+        embed = discord.Embed(
+            title=Twemoji.alias_to_name(UNICODE_EMOJI_ENGLISH[emoji]),
+            description=f"{codepoint.replace('-', ' ')}\n[Download svg]({Twemoji.get_url(codepoint, 'svg')})",
+            colour=Colours.twitter_blue,
+        )
+        embed.set_thumbnail(url=Twemoji.get_url(codepoint, "png"))
+        return embed
+
+    @staticmethod
+    def emoji(codepoint: str) -> str:
+        """
+        Returns the emoji corresponding to a given `codepoint`.
+
+        The return value is an emoji character, such as "🍂". The `codepoint`
+        argument can be of any format, since it will be trimmed automatically.
+        """
+        if code := Twemoji.trim_code(codepoint):
+            return chr(int(code, 16))
+        return ""
```

Comment:
This can be a ternary operation like this:
```suggestion
        return chr(int(code, 16)) if (code := Twemoji.trim_code(codepoint)) else ""
```

Reply:
Do you think this is better? It looks more compact, but I'm not sure the readability is quite as good.

Either way, I'll look into changing the return type to `str | None`, so this might disappear.

Reply:
I'm rejecting this for the sake or readability. Unresolved this if you have a strong opinion regarding this.